Port application logic to TierNFT component. Upgrades.

briangershon opened the pull request:
Port application logic in `<Home>` from Next.js Starter-based [D_D-Academy-lesson-5-frontend-integration-update](https://github.com/Developer-DAO/D_D-Academy-lesson-5-frontend-integration-update) repo to RainbowKit Starter-based repo.

Logic now lives in `<TierNFT>` React component.

Changes:

- Upgrade to TypeScript and `tsx`. Add type annotations to fix warnings.
- Styles were a mix of css-in-js and modules -- converted all to css-in-js.
- Upgrade to latest 1.x RainbowKit hooks
- Replace `ethers` import with `viem` import for `parseEther`
- Bring over images
- Bring over contact ABI

briangershon committed:
Bring in initial app logic into TierNFT.
Clear out boilerplate and import TierNFT component.

briangershon committed:
Add and import contract TierNFT.json
Replace parseEther from viem since no longer using ethers.

briangershon committed:
Add TypeScript annotations. Update WAGMI hooks.

briangershon committed:
Fix styling.

briangershon committed:
Bring in images

briangershon committed:
Successful mint against Mumbai-deployed
contract at 0xefde85459E38A6465878A1b2E7DcEE135420e7C0

- added env vars
- bring in Alchemy
- bring in polygonMumbai chain
- remove Zora and Goerli
- add usePrepareContractWrite hook

elPiablo reviewed (commented):
Thanks for all the work on this so far, @briangershon 🌿

briangershon committed:
Fix setting mint price.
Fix errors when wallet not connected.
Fix misnamed variable from tokenData to tokenSupply.

briangershon committed:
Replace totalSupply hack with Transfer event listener
to retrieve newly minted tokenId.
